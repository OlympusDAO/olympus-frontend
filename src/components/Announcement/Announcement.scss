.announcement-banner {
  position: sticky;
  z-index: 5;
  top: 60px;
  width: 100%;
  max-width: 500px;
  left: 100%;
  .new-network {
    display: flex;
    align-items: center;
    justify-content: space-between;
  }

  @supports not ((-webkit-backdrop-filter: none) or (backdrop-filter: none)) {
    .new-network {
      background: linear-gradient(90deg, #3a4148ee 1.58%, #333847ff 99%) !important;

      // backdrop-filter: blur(33px) !important;
      // -webkit-backdrop-filter: blur(33px) !important;
    }
  }
}

.mobile .announcement-banner {
  top: unset;
  bottom: 0 !important;
  left: 0;
  right: 0;
  max-width: 100%;
  position: fixed;
}
