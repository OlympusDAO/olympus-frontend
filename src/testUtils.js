import React from "react";
import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import { ThemeProvider } from "@material-ui/core/styles";
import CssBaseline from "@material-ui/core/CssBaseline";
import { light as lightTheme } from "./themes/light.js";
import { Web3ContextProvider } from "./hooks/web3Context";
import store from "./store.ts";
import { createMemoryHistory } from "history";
import App from "./App";
import { I18nProvider } from "@lingui/react";
import { i18n } from "@lingui/core";

const AllTheProviders = ({ children }) => {
  return (
    <Web3ContextProvider>
      <Provider store={store}>
        <BrowserRouter basename={"/"}>
          <ThemeProvider theme={lightTheme}>
            <CssBaseline />
            {children}
          </ThemeProvider>
        </BrowserRouter>
      </Provider>
    </Web3ContextProvider>
  );
};

const customRender = (ui, options) => render(ui, { wrapper: AllTheProviders, ...options });

const renderRoute = function (route) {
  const history = createMemoryHistory();
  history.push(route);
  render(
    <Web3ContextProvider>
      <Provider store={store}>
        <I18nProvider i18n={i18n}>
          <BrowserRouter basename={"/#"}>
            <App />
          </BrowserRouter>
        </I18nProvider>
      </Provider>
    </Web3ContextProvider>,
  );
};

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render, renderRoute };
