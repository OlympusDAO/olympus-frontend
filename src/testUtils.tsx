import { i18n } from "@lingui/core";
import { I18nProvider } from "@lingui/react";
import CssBaseline from "@material-ui/core/CssBaseline";
import { ThemeProvider } from "@material-ui/core/styles";
import { render, RenderOptions, RenderResult } from "@testing-library/react";
import { createMemoryHistory } from "history";
import React, { ReactElement, ReactNode } from "react";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";

import App from "./App";
import { Web3ContextProvider } from "./hooks/web3Context";
import { ReactQueryProvider } from "./lib/react-query";
import defaultStore from "./store";
import { light as lightTheme } from "./themes/light.js";

const customRender = (ui: ReactElement, store = defaultStore, options?: RenderOptions): RenderResult => {
  const ProviderWrapper = ({ children }: { children?: ReactNode }) => (
    <Web3ContextProvider>
      <ReactQueryProvider>
        <Provider store={store}>
          <I18nProvider i18n={i18n}>
            <BrowserRouter basename={"/"}>
              <ThemeProvider theme={lightTheme}>
                <CssBaseline />
                {children}
              </ThemeProvider>
            </BrowserRouter>
          </I18nProvider>
        </Provider>
      </ReactQueryProvider>
    </Web3ContextProvider>
  );
  return render(ui, { wrapper: ProviderWrapper, ...options });
};

const renderRoute = function (route: string, store = defaultStore) {
  const history = createMemoryHistory();
  history.push(route);
  return render(
    <Web3ContextProvider>
      <ReactQueryProvider>
        <Provider store={store}>
          <I18nProvider i18n={i18n}>
            <BrowserRouter basename={"/"}>
              <App />
            </BrowserRouter>
          </I18nProvider>
        </Provider>
      </ReactQueryProvider>
    </Web3ContextProvider>,
  );
};

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render, renderRoute };
