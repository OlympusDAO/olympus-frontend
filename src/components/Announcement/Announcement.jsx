import { useState } from "react";
import { useSelector } from "react-redux";
import { Box, IconButton, Paper, SvgIcon } from "@material-ui/core";
import { NETWORKS, NEWEST_NETWORK_ID } from "src/constants";
import Pill from "../Pill/Pill";
import { ReactComponent as XIcon } from "../../assets/icons/x.svg";
import "./Announcement.scss";

function Announcement() {
  const [newNetworkVisible, setNewNetworkVisible] = useState(true);

  const handleClose = () => {
    setNewNetworkVisible(false);
  };

  return (
    <div className="announcement-banner">
      {newNetworkVisible ? (
        <Paper className="new-network">
          <Box className="new-network-box" style={{ marginRight: "10px" }}>
            <Pill message="Info" style={{ color: "#F8CC82" }} />
          </Box>
          <Box className="new-network-box">
            <span>
              Treasury Stats may be inaccurate during the migration. Please check discord if you have any questions.
            </span>
          </Box>
          <Box>
            <IconButton onClick={handleClose}>
              <SvgIcon color="primary" component={XIcon} />
            </IconButton>
          </Box>
        </Paper>
      ) : (
        <div />
      )}
    </div>
  );
}

export default Announcement;
