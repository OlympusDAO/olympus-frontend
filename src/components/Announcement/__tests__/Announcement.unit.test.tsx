import { render } from "../../../testUtils";
import Announcement from "../Announcement.jsx";

describe("<Announcement/>", () => {
  it("should render component", () => {
    const { container } = render(<Announcement />);
    expect(container).toMatchSnapshot();
    expect(container).toHaveTextContent(
      "Treasury stats may be inaccurate during the migration. Please check discord if you have any questions.",
    );
  });
});
