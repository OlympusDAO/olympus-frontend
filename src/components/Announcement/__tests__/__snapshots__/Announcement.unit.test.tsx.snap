// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Announcement/> should render component 1`] = `
<div>
  <div
    class="announcement-banner ohm-card"
  >
    <div
      class="MuiPaper-root new-network MuiPaper-elevation0 MuiPaper-rounded"
      style="width: 100%;"
    >
      <div
        class="MuiBox-root MuiBox-root-1"
        style="width: 2%;"
      />
      <div
        class="MuiBox-root MuiBox-root-2"
      >
        <div
          class="MuiBox-root MuiBox-root-3 new-network-box"
          style="margin-right: 10px;"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
            focusable="false"
            viewBox="0 0 20 20"
          >
            info-fill.svg
          </svg>
        </div>
        <div
          class="MuiBox-root MuiBox-root-4 new-network-box"
        >
          <p
            class="MuiTypography-root MuiTypography-body2"
          >
            Treasury stats may be inaccurate during the migration. Please check discord if you have any questions.
          </p>
        </div>
      </div>
      <div
        class="MuiBox-root MuiBox-root-5"
      >
        <button
          class="MuiButtonBase-root MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeSmall"
              focusable="false"
              viewBox="0 0 20 20"
            >
              x.svg
            </svg>
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;
